README.md

```diff
@@ -84,6 +84,144 @@ curl http://localhost:8080/weather-agent \
    }' | jq
 ```
 
+### Testing Agent Card URL Rewriting
+
+The `agentcard-rw` plugin rewrites internal Kubernetes cluster URLs to external gateway URLs and filters out unsupported transports.
+
+#### Prerequisites
+
+Start the mock agent from `agent-samples`:
+
+```bash
+cd ./PAAL/agent-samples/wiremock/mock-agent
+make docker-run  # Starts on port 8080
+```
+
+Configure the mock agent with test data (internal cluster URLs):
+
+```bash
+curl -X POST http://localhost:8080/__admin/mappings \
+  -H 'Content-Type: application/json' \
+  -d '{
+    "request": {
+      "method": "GET",
+      "urlPath": "/.well-known/agent-card.json"
+    },
+    "response": {
+      "status": 200,
+      "headers": {"Content-Type": "application/json"},
+      "jsonBody": {
+        "name": "mock_agent",
+        "description": "Mock Agent that echoes back the input text 1:1",
+        "preferredTransport": "JSONRPC",
+        "protocolVersion": "0.3.0",
+        "capabilities": {},
+        "supportsAuthenticatedExtendedCard": false,
+        "url": "http://localhost:8080",
+        "additionalInterfaces": [
+          {"transport": "http", "url": "http://mock-agent:8080"},
```

Comment:
This does not match the transport types that the spec suggests: https://a2a-protocol.org/latest/specification/#555-agentinterface-object